libsForQt5.qtsystems: init at unstable-2019-01-03

OPNA2608 opened the pull request:
###### Description of changes

Adds the [Qt Systems module](https://github.com/qt/qtsystems) for Qt5. Will be needed for Lomiri.

~~Same as #214164, I'll give this some more checking for possible problems once I'm further along with Lomiri.~~

Looks okay to me now.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
I tested that they can run, but I don't know what they actually do or how Lomiri uses them, if at all.
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

OPNA2608 committed:
libsForQt5.qtsystems: init at unstable-2019-01-03

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1902

SuperSandro2000 commented on pkgs/development/libraries/qtsystems/default.nix:
```diff
@@ -0,0 +1,116 @@
+    description = "Qt Systems";
+    homepage = "https://github.com/qt/qtsystems";
+    license = with licenses; [ lgpl3Only /* or */ gpl2Plus ];
+    platforms = platforms.all;
```
```suggestion
    platforms = platforms.unix;
```
all other qt things are also unix

SuperSandro2000 reviewed (commented):
Why is this not under ``pkgs/development/libraries/qt-5`` ?

OPNA2608 commented:
More of the package works like this (substitution of `QT_INSTALL_EXAMPLES` by one of the Qt hooks seems borked) and `libsForQt5.qtfeedback` (which similarly doesn't have any releases) is also packaged outside of the Qt5 infrastructure.

Moved into the `pkgs/development/libraries/qt-5` infrastructure. Haven't added it to `full` because some of the other Qt projects also don't seem to be in there, and it doesn't seem important enough to get active development or releases by the Qt company. :shrug:

OPNA2608 committed:
libsForQt5.qtsystems: Move to pkgs/development/libraries/qt-5

Examples & tests dropped because they're harder to get working there, and most other modules don't bother with either.

OPNA2608 committed:
libsForQt5.qtsystems: Fix outputs on Darwin, wrap servicefw

- Neither the tools nor QML plugins are being built on Darwin, so there's nothing to put into the bin output.

- `servicefw browse` needs Qt's sqlite db plugin, so wrap it.

SuperSandro2000 commented on pkgs/development/libraries/qt-5/modules/qtsystems.nix:
```diff
@@ -0,0 +1,54 @@
+  '';
+
+  meta = {
+    maintainers = with lib.maintainers; [ OPNA2608 ];
```
Shouldn't we also set at least license and description?

OPNA2608 replied:
`meta` attributes are set by `qtModule`, the majority of Qt modules don't even have a custom `meta`.

https://github.com/NixOS/nixpkgs/blob/884e3b68be02ff9d61a042bc9bd9dd2a358f95da/pkgs/development/libraries/qt-5/qtModule.nix#L70-L76

I'm just overriding `meta.maintainers` because this evidently isn't a commonly-used & updated module, and it will be used by afew packages I'm working on adding down the line.
